docs/tutorial/codecs.txt

```diff
@@ -0,0 +1,433 @@
+               throw UnsupportedValueException::invalidEncodableValue($value);
+           }
+
+           $data = [
+               '_id' => $value->id,
+               'name' => $value->name,
+           ];
+
+           if ($value->address) {
+               $data['address'] = $this->addressCodec->encode($value->address);
+           }
+
+           return MongoDB\BSON\Document::fromPHP($data);
+       }
+   }
+
+Codec Libraries
+---------------
+
+If you have a number of codecs that you want to use in multiple places, you can create a codec library. A codec library
+contains a list of codecs and checks each codec if it supports a value. If it does, it will use that codec to encode or
+decode the given value. The following code snippet changes the ``PersonCodec`` to use a codec library instead of a
+hard-coded ``DateTimeCodec``:
+
+.. code-block:: php
+
+   <?php
+
+   final class PersonCodec implements MongoDB\Codec\DocumentCodec
+   {
+       public function __construct(
+           private readonly MongoDB\Codec\CodecLibrary $codecLibrary,
+       ) {}
+
+       // Other code omitted for brevity
+       public function decode($value): Person
+       {
+           if (! $this->canDecode($value)) {
+               throw UnsupportedValueException::invalidDecodableValue($value);
+           }
+
+           $person = new Person($value->get('name'));
+           $person->id = $value->get('_id');
+           $person->createdAt = $this->codecLibrary->decode($value->get('createdAt'));
+
+           return $person;
+       }
+
+       public function encode($value): MongoDB\BSON\Document
+       {
+           if (! $this->canEncode($value)) {
+               throw UnsupportedValueException::invalidEncodableValue($value);
+           }
+
+           return MongoDB\BSON\Document::fromPHP([
+               '_id' => $value->id,
+               'name' => $value->name,
+               'createdAt' => $this->codecLibrary->encode($value->createdAt),
+           ]);
+       }
+   }
+
+This way, you can configure a global codec library and reuse it in multiple places:
+
+.. code-block:: php
+
+   <?php
+
+    $codecLibrary = new MongoDB\Codec\CodecLibrary([
+         new DateTimeCodec(),
+    ]);
+
+    $personCodec = new PersonCodec($codecLibrary);
+
+.. note::
+
+   Be careful when adding ``DocumentCodec`` instances to a codec library. The codec library will only use the first
+   codec that supports a given value. While this will most likely work fine when encoding to BSON, it can lead to issues
+   when decoding from BSON as you'll have to inspect the BSON document to detect whether the codec supports it. For this
+   reason, the ``codec`` option in a collection does not accept a codec library.
+
+Library-aware Codecs
+~~~~~~~~~~~~~~~~~~~~
+
+If you have a codec that needs to be aware of the codec library, you can implement the
+``MongoDB\Codec\KnowsCodecLibrary`` interface. When the codec is added to a library, this library is then injected into
```

Comment:
Did we discuss calling this CodecLibraryAware in some previous PR? I don't feel strongly either way but seeing the section title here made me wonder why it's named KnowsCodecLibrary.

Reply:
Looking through #1125 I couldn't find anything related to it. That said, I do remember the discussion and I vaguely remember us agreeing that `CodecLibraryAware` was better, but I may have forgotten to change it. @GromNaN I believe it was you that suggested the alternative, do I remember correctly?

Reply:
Codec libraries were removed from the code and in this tutorial.